internal/internal_command_state_machine.go

```diff
@@ -1253,10 +1259,18 @@ func (h *commandsHelper) recordMutableSideEffectMarker(mutableSideEffectID strin
 	return command
 }
 
-func (h *commandsHelper) startChildWorkflowExecution(attributes *commandpb.StartChildWorkflowExecutionCommandAttributes) commandStateMachine {
+// startChildWorkflowExecution can return an error in the event that there is already a child wf
+// with the same ID which exists as a command in memory. Other SDKs actually will send this command
+// to server, and have it reject it - but here the command ID is exactly equal to the child's wf ID,
+// and changing that without potentially blowing up backwards compatability is difficult. So we
+// return the error eagerly locally, which is at least an improvement on panicking.
+func (h *commandsHelper) startChildWorkflowExecution(attributes *commandpb.StartChildWorkflowExecutionCommandAttributes) (commandStateMachine, error) {
 	command := h.newChildWorkflowCommandStateMachine(attributes)
+	if h.commands[command.getID()] != nil {
+		return nil, &childWorkflowExistsWithId{id: attributes.WorkflowId}
```

Comment:
I can't say I love this, but as is often the case in the Go SDK it seemed like the least offensive of some unfortunate options.